controls/V-72025.rb

```diff
@@ -48,40 +41,41 @@
 -rw-r--r-- 1 smithj smithj 193 Mar  5 17:06 file2
 -rw-r--r-- 1 smithj sa        231 Mar  5 17:06 file3
 
-If any files are found with an owner different than the group home directory user,
-check to see if the user is a member of that group with the following command:
+If any files are found with an owner different than the group home directory
+user, check to see if the user is a member of that group with the following
+command:
 
 # grep smithj /etc/group
 sa:x:100:juan,shelley,bob,smithj
 smithj:x:521:smithj
 
 If the user is not a member of a group that group owns file(s) in a local
 interactive user’s home directory, this is a finding."
-  tag "fix": "Change the group of a local interactive user’s files and directories
-to a group that the interactive user is a member of. To change the group owner of a
-local interactive user’s files and directories, use the following command:
+  tag "fix": "Change the group of a local interactive user’s files and
+directories to a group that the interactive user is a member of. To change the
+group owner of a local interactive user’s files and directories, use the
+following command:
 
 Note: The example will be for the user smithj, who has a home directory of
 \"/home/smithj\" and is a member of the users group.
 
 # chgrp users /home/smithj/<file>"
+  tag "fix_id": "F-78377r1_fix"
 
-  # Assumption - users' home directories created in "home"
-  home_dirs = command('ls -d /home/*').stdout.split("\n")
-  home_dirs.each do |home|
-    home_files = command("find #{home} ! -name '.*'").stdout.split("\n")
-    home_user = home.split("/")
-    user_groups = command("groups #{home_user[2]}").stdout.split(" ")
-    user_groups.delete_at(0)
-    user_groups.delete_at(0)
-    home_files.each do |curr_file|
-      describe.one do
-        user_groups.each { |curr_group|
-          describe file(curr_file) do
-            its('group') { should cmp "#{curr_group}" }
-          end
-        }
-      end
-    end
+  IGNORE_SHELLS = NON_INTERACTIVE_SHELLS.join('|')
+
+  findings = Set[]
+  users.where{ !shell.match(IGNORE_SHELLS) && (uid >= 1000 || uid == 0)}.entries.each do |user_info|
+    next if EXEMPT_HOME_USERS.include?("#{user_info.username}")
+    find_args = ""
+    user_info.groups.each { |curr_group|
+      # some key files and secure dirs (like .ssh) are group owned 'root'
+      find_args = find_args + "-not -group #{curr_group} -o root"
+    }
+    findings = findings + command("find #{user_info.home} #{find_args}").stdout.split("\n")
```

Comment:
See `find` comment above. Many home directories are in `autofs` and you will potentially destroy networks doing this.